docs/guides/mysql/backup/kubestash/customization/index.md

```diff
@@ -0,0 +1,444 @@
+
+## Customizing Restore Process
+
+KubeStash also uses `mysql` during the restore process. In this section, we are going to show how you can pass arguments to the restore process, restore a specific snapshot, run restore job as a specific user, etc.
+
+
+### Passing arguments to the restore process
+
+Similar to the backup process, you can pass arguments to the restore process through the `args` params under `addon.tasks[*].params` section. 
+
+This example will restore data from database `testdb` only.
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: BackupConfiguration
+metadata:
+  name: sample-mysql-backup
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: MySQL
+    namespace: demo
+    name: sample-mysql
+  backends:
+    - name: gcs-backend
+      storageRef:
+        namespace: demo
+        name: gcs-storage
+      retentionPolicy:
+        name: demo-retention
+        namespace: demo
+  sessions:
+    - name: frequent-backup
+      scheduler:
+        schedule: "*/5 * * * *"
+        jobTemplate:
+          backoffLimit: 1
+      repositories:
+        - name: gcs-mysql-repo
+          backend: gcs-backend
+          directory: /mysql
+          encryptionSecret:
+            name: encrypt-secret
+            namespace: demo
+      addon:
+        name: mysql-addon
+        tasks:
+          - name: logical-backup
+            params:
+              args: --one-database=testdb
+```
+
+### Restore specific snapshot
+
+You can also restore a specific snapshot. At first, list the available snapshot as bellow,
+
+```bash
+➤ kubectl get snapshots.storage.kubestash.com -n demo -l=kubestash.com/repo-name=gcs-mysql-repo
+NAME                                                            REPOSITORY       SESSION           SNAPSHOT-TIME          DELETION-POLICY   PHASE       AGE
+gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725257849   gcs-mysql-repo   frequent-backup   2024-09-02T06:18:01Z   Delete            Succeeded   15m
+gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725258000   gcs-mysql-repo   frequent-backup   2024-09-02T06:20:00Z   Delete            Succeeded   13m
+gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725258300   gcs-mysql-repo   frequent-backup   2024-09-02T06:25:00Z   Delete            Succeeded   8m34s
+gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725258600   gcs-mysql-repo   frequent-backup   2024-09-02T06:30:00Z   Delete            Succeeded   3m34s
+```
+
+The below example shows how you can pass a specific snapshot name in `.dataSource` section.
+
+```yaml
+apiVersion: core.kubestash.com/v1alpha1
+kind: RestoreSession
+metadata:
+  name: restore-sample-mysql
+  namespace: demo
+spec:
+  target:
+    apiGroup: kubedb.com
+    kind: MySQL
+    namespace: demo
+    name: restored-mysql
+  dataSource:
+    repository: gcs-mysql-repo
+    snapshot: latest
```

Comment:
we should provide value for snapshot, right?

`snapshot: gcs-mysql-repo-sample-mysql-backup-frequent-backup-1725258600`